tests/pipelines/lint/test_rocrate_readme_sync.py

```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
+
+from ..test_lint import TestLint
+
+
+class TestLintROcrateReadmeSync(TestLint):
+    def test_rocrate_readme_sync_pass(self):
+        self.lint_obj._load()
+        results = self.lint_obj.rocrate_readme_sync()
+        assert len(results.get("warned", [])) == 0
+        assert len(results.get("failed", [])) == 0
+        assert len(results.get("passed", [])) > 0
+
+    def test_rocrate_readme_sync_fail(self):
+        self.lint_obj._load()
+
+        json_path = Path(self.lint_obj.wf_path, "ro-crate-metadata.json")
+        with open(json_path, "w") as f:
+            f.write('{ "@graph": [{"description": "This is a test script"}] }')
```

Comment:
To parse the file as a JSON you can use the library `json` as you did in `nf_core/pipelines/lint/rocrate_readme_sync.py`. This code is untested so make sure that I didn't make any mistake before accepting the suggestion 😄 
```suggestion
        with open(json_path, "rw") as f:
            try:
                rocrate = json.load(fh)
            except json.JSONDecodeError as e:
                raise UserWarning(f"Unable to load JSON file '{json_path}' due to error {e}")
            rocrate["@graph"][0]["description"] = "This is a test script"
            json.dump(rocrate, fh, indent=4)
```